### ISSUE TITLE
Documentation: default config given in README has the wrong name.

### ISSUE THREAD

**Issue by jcronk** [troessner/reek#1327]
In the README, it says that the default name for the config file is ".reek.yml", but apparently it's anything ending in ".reek".

**Comment by troessner** [comment]
Well, that's what we get for doing breaking changes in order to [release Reek v5](https://github.com/troessner/reek/issues/1305) and being to lazy to update 2 separate sets of documentation 😆 
@mvz any chance we can accelerate those 2 PR's:

* [Update app configuration api](https://github.com/troessner/reek/pull/1310)
* [Use strings not regexes in our configuration file](https://github.com/troessner/reek/pull/1322)

before more users are confused by our docs?

If you don't have time right now we could just revert the changes to our README and /docs, get the release going and then have all of those doc changes released in the scope the [Prepare Reek 5 release](https://github.com/troessner/reek/pull/1324)

**Comment by mvz** [comment]
Sadly, the common way for people to read the README for Ruby stuff is to go the `master` branch on GitHub, rather than reading it from the installed gem.

Many projects adopt the solution of putting 'this is the README for master' in the README, but I think that's problematic because you need to remove it for every release.

Finally, it's not like breaking changes don't usually happen so why are people not used to switching to the relevant release tag?

TL;DR: I'm sorry that people get confused but they should be encouraged to read the README that comes with the installed gem, and gem authors shouldn't need to maintain two sets of documentation.

**Comment by mvz** [comment]
I dug into this some more, and ideally I would want people to do one of these two things:

1. Run `ri reek:README` from the command line.
2. Go to https://www.rubydoc.info/gems/reek.

There are problems with both of these.

Option 1 doesn't work because somehow `ri` files are generated for `CHANGELOG.md` and `License.txt`, but not for `README.md`.

Option 2 is better, but the hand-crafted table of contents doesn't work there, you have to use the generated one on the right. Also rubydoc.info lacks an option to quickly switch to another version of the same gem.

**Comment by troessner** [comment]
>> Sadly, the common way for people to read the README for Ruby stuff is to go the master branch on GitHub, rather than reading it from the installed gem. 

I am doing the exact same thing. 

>> TL;DR: I'm sorry that people get confused but they should be encouraged to read the README that comes with the installed gem, and gem authors shouldn't need to maintain two sets of documentation.

But the current world is different and we're not going to change that (even though I agree that your approach is the cleaner one).
I think in this case the user is not to blame. Instead, we should have handled this better (a simple reek-5 release branch would have prevented all of this).

I created #1328 to fix our documentation - thanks @jcronk for reporting this!

**Comment by mvz** [comment]
> But the current world is different and we're not going to change that

That's definitely true.

> Instead, we should have handled this better (a simple reek-5 release branch would have prevented all of this).

Such a branch carries its own set of problems. I still think assembling Reek 5 on master was the right decision. However, I also agree the user is not to blame and I should have pointed that out in my "I wish the world were a different place" story above.

I'll comment on the #1329 solution there.
